Sources/BCRunner/BytecodeRunner.swift

```diff
@@ -19,6 +19,53 @@ public func runPathBytecode(_ path: CGMutablePath, fromData data: Data) throws {
   }
 }
 
+public func runMergedBytecode(
+  fromData data: Data,
+  _ context: CGContext,
+  _ startIndex: Int,
+  _ endIndex: Int
+) throws {
+  let sz = data.count
+  try data.withUnsafeBytes {
+    let ptr = $0.baseAddress!.assumingMemoryBound(to: UInt8.self)
+    let key = NSString(cString: ptr, encoding: String.Encoding.ascii.rawValue)!
+    var decompressedArray = cache.value(key: key) ?? []
+
+    if decompressedArray.isEmpty {
+      decompressedArray = try BytecodeCompressor.decompressBytecode(ptr, sz)
+      cache.insert(key: key, value: decompressedArray)
+    }
+
+    let partArray = Array(decompressedArray[startIndex...endIndex])
+    try BytecodeRunner.run(context, partArray, partArray.count)
+  }
+}
+
+@_cdecl("runMergedBytecode")
+@available(macOS 10.15, *)
+public func runMergedBytecode(
+  _ context: CGContext,
+  _ start: UnsafePointer<UInt8>,
+  _ len: Int,
+  _ startIndex: Int,
+  _ endIndex: Int
+) {
+  do {
+    let key = NSString(cString: start, encoding: String.Encoding.ascii.rawValue)!
```

Comment:
May be it's better here to use `NSNumber`? Seems more natural, as UnsafePointer is hashable already and the only reason we converting it to string is that NSCache requires NSObject's hash.